.github/workflows/ci.yml

```diff
@@ -64,13 +75,19 @@ jobs:
       - uses: actions/checkout@c85c95e3d7251135ab7dc9ce3241c5835cc595a9 # v3.5.3
         with:
           fetch-depth: 0
+      - uses: actions/setup-node@64ed1c7eab4cce3362f8c340dee64e5eaeef8f7c # tag=v3.6.0
+        with:
+          node-version-file: '.nvmrc'
+          cache: 'yarn'
       - uses: actions/cache@88522ab9f39a2ea568f7027eddc7d8d8bc9d59c8 # tag=v3.3.1
         with:
           enableCrossOsArchive: true
+          # TODO: setup `build --repository_cache` flag and cache the result
           path: |
-            .yarn/cache
-            .yarn/install-state.gz
-          key: v1-${{hashFiles('yarn.lock')}}
+            %AppData%/bazelisk
+          key: ${{ github.job }}-${{ runner.os }}-${{ hashFiles('**/.bazelversion') }}
+          restore-keys: |
+            ${{ github.job }}-${{ runner.os }}-
```

Comment:
One final nit: I think we can actually remove this as there is no incremental use if e.g. no cache hit was for Bazel 6.0.0- it won't help if we restore 5.0.0 then- it would rather just bloat the cache over time

Reply:
Good call